client/chat/user-overlay.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { CSSTransition } from 'react-transition-group'
+import styled from 'styled-components'
+
+import Avatar from '../avatars/avatar.jsx'
+import Popover from '../material/popover.jsx'
+import { fastOutSlowIn } from '../material/curve-constants.js'
+import { Title, singleLine } from '../styles/typography'
+
+const transitionNames = {
+  appear: 'enter',
+  appearActive: 'enterActive',
+  enter: 'enter',
+  enterActive: 'enterActive',
+  exit: 'exit',
+  exitActive: 'exitActive',
+}
+
+const Contents = styled.div`
+  min-width: 256px;
+`
+
+const Header = styled.div`
+  position: relative;
+  display: flex;
+  justify-content: center;
+  padding-top: 12px;
+
+  .enter & {
+    opacity: 0;
+    transform: translateY(-16px);
+    transition-property: all;
+    /* TODO(tec27): this should be linear for opacity, and probably linearOutSlowIn for
+       translation? */
+    transition-timing-function: ${fastOutSlowIn};
+  }
+
+  .enterActive & {
+    opacity: 1;
+    transform: translateY(0px);
+    pointer-events: none;
+  }
+
+  .exit & {
+    opacity: 1;
+    transition-property: opacity;
+    transition-timing-function: linear;
+  }
+
+  .exitActive & {
+    opacity: 0;
+    pointer-events: none;
+  }
+`
+
+const StyledAvatar = styled(Avatar)`
+  width: 32px;
+  height: 32px;
+  margin-bottom: 8px;
+`
+
+const Username = styled(Title)`
+  margin-top: 0;
+  margin-bottom: 0;
+  padding-left: 8px;
+  ${singleLine};
+`
+
+const Actions = styled.div`
+  position: relative;
+  padding-top: 8px;
+  padding-bottom: 8px;
+
+  .enter & {
+    opacity: 0;
+    transform: translateY(-16px);
+    transition-property: all;
+    /* TODO(tec27): this should be linear for opacity, and probably linearOutSlowIn for
+       translation? */
+    transition-timing-function: ${fastOutSlowIn};
+  }
+
+  .enterActive & {
+    opacity: 1;
+    transform: translateY(0);
+    pointer-events: none;
+  }
+
+  .exit & {
+    opacity: 1;
+    transition-property: opacity;
+    transition-timing-function: linear;
+  }
+
+  .exitActive & {
+    opacity: 0;
+    pointer-events: none;
+  }
+`
+
+export default class UserOverlay extends React.Component {
```

Comment:
Depending on the final design of the user overlay component, I feel like there's a lot of code here that could be reused from the self profile overlay component ^^.

Either by using a single component for both overlays (if the designs end up being _very_ similar), or pulling common parts in some generic component and using that component in both overlays.